### DISCUSSION TITLE
How to set `routeRules` swr cache only for logged out users?

### DISCUSSION THREAD

**Discussion by meirroth** [nuxt/nuxt#27232] [Questions]
Hey Nuxters!

I'm trying to speed up TTFB on our homepage and thought of using SWR via [`routeRules`](https://nitro.unjs.io/config#routerules):

```js
routeRules: {
  '/': { swr: 60 * 60 * 1000 },
}
```

But this causes logged in users to see the logged out version and displays the wrong language (using @nuxtjs/i18n).

I'm wondering if there's a way to dynamically cache based on auth and locale cookies.

Any ideas on how to solve this?

Thanks!

Meir

Related issues:
- https://github.com/nuxt/nuxt/discussions/16078
- https://github.com/unjs/nitro/issues/1544

**Comment by TheAlexLichter** [comment]
You can track that feature in https://github.com/nuxt/nuxt/issues/26598

**Reply by meirroth** [reply]
@manniL got this message 5 times 😅 anyway, thank you. Love your videos btw.

**Reply by TheAlexLichter** [reply]
Sorry, Airplan wifi 🙈


Happy to hear you enjoy them 🙏🏻

**Reply by meirroth** [reply]
@manniL No worries. Have a pleasant flight!

How do you suppose the Nitro hook you proposed over there will resolve this issue?
And I'm curious if someone has found a solution with what's currently available.

**Reply by meirroth** [reply]
@manniL Also, maybe we can leave this open in case others have a solution?

**Reply by Plinpod** [reply]
Hey, I'm looking to do the same thing. I'm using `sidebase/nuxt-auth` for my authentication and I'd like to bypass the cache entirely for certain pages if the user is logged in.

Wasn't sure how the feature mentioned above correlated to page caching. Thanks!

**Reply by meirroth** [reply]
@Plinpod I never got an answer for this :(

**Reply by meirroth** [reply]
@TheAlexLichter Hey, any chance you can reopen this item? Maybe someone from the community has a solution now?

**Reply by TheAlexLichter** [reply]
@meirroth They can post that in https://github.com/nuxt/nuxt/issues/26598 (which should have better visibility 👍🏻)
https://nuxt-multi-cache.dulnan.net/ might be an option!
